mscore/plugin/api/elements.cpp

```diff
@@ -69,6 +69,53 @@ void Note::setTpc(int val)
             set(Pid::TPC2, val);
       }
 
+
+//---------------------------------------------------------
+//   Note::setOnTimeOffset
+//---------------------------------------------------------
+
+void Note::setOnTimeOffset(int v)
+      {
+      Ms::Score* score = Ms::MuseScoreCore::mscoreCore->currentScore();
+      if (!score) {
+            qWarning("PluginAPI::Note::setOnTimeOffset: A score is required.");
+            return;
+            }
+      NoteEvent* event = note()->noteEvent(0);
```

Comment:
Why do you use exactly the first note event (both here and in libmscore)? Some notes may have more than one note event, for example notes with ornaments applied. If your intention is to add an ability to allow a more complete control over playback for plugins then it may be worth to expose play events to plugins directly in some way. I am not sure how would it be better to implement that but otherwise only a limited functionality will be exposed, and it is not clear in which way that functionality should actually be limited.

Reply:
A great question!

I patterned the code after the piano roll editor. However that code deals with PianoItem's which refer to single notes. As for the implementation this discussion drove the work:

https://musescore.org/en/node/291414
https://musescore.org/en/node/291519

This is where your answer would be found. I'll present your concerns to via creating an issue/feature request to link it all together.

-Dale

I'm a code monkey for this task. Reworking the above simple api may be required to meet it so

Reply:
https://musescore.org/en/node/291708

Reply:
>I'm sending the discussion there.

Or is there a better way to indicate active discussion of the PR? 

I'm not sure if all issue contributors also access github.

-Dale

Reply:
Hi @dmitrio95 

Since you raised the issue increasing uncertainty it seems fair that you should engage in the discussion to reduce it or suggest an architect join it.

https://musescore.org/en/comment/931859

I'd rather not commit to more work if it's not going to be accepted.

-Dale

Reply:
OK, I'll look through the discussion then. At first glance the best solution I see is to expose `NoteEvent`s to plugins in some way, but maybe there are some other considerations on that issue that I don't see yet. Anyway, I'll comment in that issue.

Reply:
Excellent! Thanks!

-Dale